docs/ops/sort/MatrixNMS_8.md

```diff
@@ -0,0 +1,167 @@
+##  MatrixNonMaxSuppression<a name="MatrixNonMaxSuppression"></a> {#openvino_docs_ops_sort_MatrixNms_8}
+
+**Versioned name**: *MatrixNonMaxSuppression-8*
+
+**Category**: *Sorting and maximization*
+
+**Short description**: *MatrixNonMaxSuppression* performs matrix non maximum suppression (NMS) of the boxes with predicted scores.
+
+**Detailed description**: *MatrixNonMaxSuppression* is a multi-phase operation.
+
+1. First this operation selects candidate bounding boxes with scores higher than `score_threshold`.
+2. Second for each class at most `nms_top_k` candidate boxes are selected.
+3. Score of the candidate boxes are then decayed according to Matrix NMS algorithm [Wang et al](https://arxiv.org/abs/2003.10152.pdf). This algorithm is applied independently to each class and each batch element. Boxes of `background_class` are skipped and thus eliminated during the process.
+4. Finanly, the operation selects boxes with the decayed scores higher than `post_threshold`, and selects at most `keep_top_k` scoring candidate boxes per batch element.
+
+The Matrix NMS algorithm is described below:
+1.  Sort descending the candidate boxes by score, and compute `n*n` pairwise IOU (IntersectionOverUnion) matrix `X` for the top `n` boxes. Suppose `n` is the number of candidate boxes.
+2.  Set the lower triangle and diagonal of `X` to 0. Therefore get the upper trianglar matrix `X`.
+3.  Take th column-wise max of `X` to compute a vector `K` of maximun IOU for each candidate box.
+4.  Repeat element value of `K` along axis 1. Suppose this get a matrix `X_cmax`.
+5.  Compute the decay factor: `decay_factor = exp((X_cmax**2 - X**2) * gaussian_sigma)` if `decay_function` is `guassian`, else `decay_factor = (1 - X) / (1 - X_cmax)`.
+6.  Take the column-wise min of `decay_factor`, and element-wise multiply with scores to decay them.
+
+**Attributes**:
+
+* *sort_result*
+
+  * **Description**: *sort_result* specifies the order of output elements.
+  * **Range of values**: `class`, `score`, `none`
+    * *class* - sort selected boxes by class id (ascending).
+    * *score* - sort selected boxes by score (descending).
+    * *none* - do not guarantee the order.
+  * **Type**: `string`
+  * **Default value**: `none`
+  * **Required**: *No*
+
+* *sort_result_across_batch*
+
+  * **Description**: *sort_result_across_batch* is a flag that specifies whenever it is necessary to sort selected boxes across batches or not.
```

Comment:
PaddlePaddle has an attribute called normalized to specify whether the bounding boxes are normalized by the image width/height. Do we need to add this attribute ?

Reply:
Don't have to be in OV Op. We will handle it in our PDPD->OV conversion.

Reply:
How do you know the image size if you want to handle it in PDPD->OV conversion? So I suggest to add this attribute.

Reply:
During op conversion, it gets the attribute value True/False, but during calculation, they have different IOU calculation methods depends on the normalized value. how can we handle this during the Ops conversion ?

Reply:
This "normalize" attribute is PDPD specific, no need to use image size. It simply add a const offset to the original bbox. And as discussed with ZhangYi, we can handle this with a decomposition. 

Please refer to Paddle2ONNX, which solve this by adding 2 Add operation before NMS. 
https://github.com/PaddlePaddle/Paddle2ONNX/blob/e95695af98f1cf417d3ba2c0e81abf7b59948ab6/paddle2onnx/op_mapper/detection/multiclass_nms.py#L106

Reply:
@lazarevevgeny I am so sorry I have to reopen this discussion. During test I found it cannot be solved in decomposition. Extra substract operation is required after NMS. It looks awkward. What's more, the extra substracts won't  success as they are dynamic shape input.